japecheck/analyzer.go

```diff
@@ -351,6 +366,52 @@ func parseServerRoute(kv *ast.KeyValueExpr, pass *analysis.Pass) (*serverRoute,
 		return nil, false
 	}
 
+	checkCfg := func(cfgg *cfg.CFG) {
+		isCheckCall := func(pass *analysis.Pass, n ast.Node) bool {
+			if call, ok := n.(*ast.CallExpr); !ok {
+				return false
+			} else if sel, ok := call.Fun.(*ast.SelectorExpr); !ok {
+				return false
+			} else if typ := typeof(sel.X); typ == nil || typ.String() != "go.sia.tech/jape.Context" {
+				return false
+			} else if sel.Sel.Name != "Check" {
+				return false
+			}
+			return true
+		}
+		checkSuccessor := func(block *cfg.Block) bool {
+			if len(block.Nodes) == 1 {
+				if _, ok := block.Nodes[0].(*ast.ReturnStmt); ok {
+					return true
+				}
+			}
+			return false
+		}
+		for _, block := range cfgg.Blocks {
+			for _, n := range block.Nodes {
+				if e, ok := n.(*ast.BinaryExpr); ok {
+					// check if jc.Check() != nil
+					if isCheckCall(pass, e.X) && e.Op == token.NEQ {
+						// check that successor is just a return statement
+						if !checkSuccessor(cfgg.Blocks[block.Succs[0].Index]) {
+							pass.Report(analysis.Diagnostic{
+								Pos:     e.Pos(),
+								Message: fmt.Sprintf("jc.Check() != nil doesn't immediately return"),
+							})
+						}
+					}
+				}
+			}
+		}
+	}
+
+	cfgs := pass.ResultOf[ctrlflow.Analyzer].(*ctrlflow.CFGs)
+	if fd != nil {
+		checkCfg(cfgs.FuncDecl(fd))
+	} else if fl != nil {
+		checkCfg(cfgs.FuncLit(fl))
+	}
```

Comment:
let's move this check into its own function, rather than doing it in `parseServerRoute`